manager/integration/tests/test_csi_snapshotter.py

```diff
@@ -1207,6 +1218,123 @@ def finalizer():
 
     request.addfinalizer(finalizer)
 
+def test_csi_volumesnapshot_backing_image_with_selectors(client, # NOQA
+                                                         core_api, # NOQA
+                                                         csi_pv, # NOQA
+                                                         pod_make, # NOQA
+                                                         pvc, # NOQA
+                                                         request, # NOQA
+                                                         volume_name, # NOQA
+                                                         volumesnapshotclass, # NOQA
+                                                         volumesnapshot): # NOQA
+    """
+    Test Create BackingImage with nodeSelector and diskSelector
+    using VolumeSnapshot with a given Volume
+
+    This test is to test users can set the nodeSelector and diskSelector
+    through csi volumeSnapshot.
+    For the scheduling, we already test it in backing image e2e tests.
+
+    Setup
+    - Create a VolumeSnapshotClass with type `bi`
+        ```
+        kind: VolumeSnapshotClass
+        apiVersion: snapshot.storage.k8s.io/v1
+        metadata:
+            name: longhorn-snapshot-vsc
+        driver: driver.longhorn.io
+        deletionPolicy: Delete
+        parameters:
+            type: bi
+            nodeSelector: "node1"
+            diskSelector: "ssd"
+        ```
+
+    Given
+    - The Volume test-vol attached to a workload,
+      having data and computed md5sum.
+    - Create the PV and PVC from the Volume
+
+    When
+    - Creating the VolumeSnapshot
+        ```
+        apiVersion: snapshot.storage.k8s.io/v1
+        kind: VolumeSnapshot
+        metadata:
+            name: test-snapshot-backing
+        spec:
+            volumeSnapshotClassName: longhorn-snapshot-vsc
+            source:
+                persistentVolumeClaimName: test-vol
+        ```
+
+    Then
+    - A BackingImage is created with the correct selectors
```

Comment:
It's better to verify the BI data as well.

Reply:
The content verification will be perform in the following test
`test_csi_volumesnapshot_backing_image_basic`
to save the testing time, I prefer focus on the selectors parameters passing through csi in this this.